### PR TITLE
Data source improvements: enable/disable data sources

### PR DESCRIPTION
**Pull request by AndrewLngdn**
@jleealpine Do you have time to take a look at this? QA wants this stuff merged into master soon

### CODE REVIEW

**Comment by alpinebuilder** [comment]
Can one of the admins verify this patch?

File: app/models/data_source.rb [hunk 1]
```diff
@@ -164,11 +165,11 @@ def refresh(options={})
   end
 
   def refresh_databases_later
-    SolrIndexer.SolrQC.enqueue_if_not_queued('DataSource.refresh_databases', id) unless being_destroyed?
+    SolrIndexer.SolrQC.enqueue_if_not_queued('DataSource.refresh_databases', id) unless being_destroyed? || disabled?
```
Comment (AndrewLngdn) [inline app/models/data_source.rb]: check being_destroyed to see if it attempts DS connection

File: app/presenters/search_presenter.rb [hunk 2]
```diff
@@ -40,6 +40,11 @@ def to_hash
     }.merge(workspace_specific_results)
   end
 
+
+  def self.filter_disabled(data_source_results)
+    data_source_results.select{|ds|ds.state != 'disabled'} unless Permissioner.is_admin?(current_user)
```
Comment (AndrewLngdn) [inline app/presenters/search_presenter.rb]: this should use .disabled?

File: spec/models/data_source_spec.rb [hunk 3]
```diff
@@ -284,6 +301,12 @@
       mock(SolrIndexer.SolrQC).enqueue_if_not_queued("DataSource.reindex_data_source", data_source.id)
       data_source.solr_reindex_later
     end
+
+    it "should not enqueue a job if the data source is disabled" do
+      data_source.update_attributes(:state => 'disabled')
```
Comment (AndrewLngdn) [inline spec/models/data_source_spec.rb]: use dont_allow workaround

File: spec/models/hdfs_data_source_spec.rb [hunk 4]
```diff
@@ -78,6 +78,16 @@
   describe "#check_status!" do
     let(:data_source) { hdfs_data_sources(:hadoop) }
 
+    context "when the data source is disabled" do
+      it "does not check the status" do
+        data_source.update_attributes(:state => 'disabled')
+        any_instance_of(HdfsDataSource) do |ds|
+          mock(ds).disabled?
```
Comment (AndrewLngdn) [inline spec/models/hdfs_data_source_spec.rb]: use dont_allow workaround

**Comment by AndrewLngdn** [comment]
Reviewed by @midnighteuler

**Pull request merged**